composeApp/src/commonMain/kotlin/com/rahulrav/camera/scan/CameraScanViewModel.kt

```diff
@@ -0,0 +1,89 @@
+package com.rahulrav.camera.scan
+
+import SonyCameraControl
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.snapshots.SnapshotStateList
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import co.touchlab.kermit.Logger
+import com.juul.kable.Advertisement
+import getPlatform
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.IO
+import kotlinx.coroutines.launch
+
+interface CameraScanViewModel {
+    val state: State<ScanState>
+    
+    fun doScan()
+
+    fun pair(camera: DiscoveredCamera)
+}
+
+class CameraScanViewModelImpl(
+    private val cameraControl: SonyCameraControl = SonyCameraControl(getPlatform()),
+    private val scanDispatcher: CoroutineDispatcher = Dispatchers.IO,
+) : ViewModel(), CameraScanViewModel {
+
+    private val _state = mutableStateOf<ScanState>(ScanState.StartingScan)
```

Comment:
You are using Kotlin 2.0, you can use the `field` keyword. 😄

Reply:
![image](https://github.com/user-attachments/assets/d886b77f-ff4e-4ded-a49f-43d293d9ec8f)
:((

Reply:
We'd need to switch to K2 first...